### DISCUSSION TITLE
[IntelliSense] Create package `monaco-tailwind`

### DISCUSSION THREAD

**Discussion by remcohaszing** [tailwindlabs/tailwindcss#7889] [Ideas]
Currently https://github.com/tailwindlabs/tailwindcss-intellisense contains 3 packages:

- [tailwindcss-language-server](https://github.com/tailwindlabs/tailwindcss-intellisense/tree/master/packages/tailwindcss-language-server)
- [tailwindcss-language-service](https://github.com/tailwindlabs/tailwindcss-intellisense/tree/master/packages/tailwindcss-language-service)
- [vscode-tailwindcss](https://github.com/tailwindlabs/tailwindcss-intellisense/tree/master/packages/vscode-tailwindcss)

There is also a [tailwind playground](https://github.com/tailwindlabs/play.tailwindcss.com) which integrates the language service into Monaco editor.

I’m interested in having tailwind integration in Monaco editor as well. Are you interested in publishing this as a separate package (`monaco-tailwindcss`)? I imagine this package would exist in the [tailwindcss-intellisense](https://github.com/tailwindlabs/tailwindcss-intellisense) repository.

If you’re interested, I’m willing to implement this.

**Comment by mhsdesign** [comment]
i would really like to see that. Im currently trying to figure out how to copy just the functionality from the playground - if you like i will help you.

**Reply by remcohaszing** [reply]
I’ve started working on this myself. The code from the playground isn’t very reusable, so I basically rewrote it from scratch reusing the same principles I used in [`monaco-yaml`](https://github.com/remcohaszing/monaco-yaml).

I think I can create an initial release this friday if I create this as a separate project, so I suggest not to spend too much time figuring out how to use the functionality from the playground. :wink:

**Reply by mhsdesign** [reply]
WOW this sounds amazing. ♥️
I figured by your github that youre pretty much allknowing what concerns those things ^^ - nevertheless i would be happy to help!

**Reply by remcohaszing** [reply]
Haha I wouldn’t say allknowing :sweat_smile: 

Anyway, I published my initial attempt at https://github.com/remcohaszing/monaco-tailwindcss. I didn’t manage to wrap it up nicely yet, so I didn’t publish to npm yet.

A demo can be started by cloning the repository, then running:

```sh
npm install
npm --workspace demo start
```

I’m open to any sort of contributions, for example:

- Bug reports
- Feature requests
- More representative Demo model updates
- A fix for the weird layout change of the demo editor instance causing page scrolling
- Upstream fixes for missing types in `@types/tailwindcss`
- A build script (modeled after `monaco-yaml`)
- Code reviews (I’ll use pull requests if you’re interested in doing code reviews)
- Anything you deem useful :)

**Reply by mhsdesign** [reply]
this is lit ^^ you rock!

lets transfer our conversation to this place ^^

**Comment by dinhanhthi** [comment]
Do you have any example of how ot integrate your package into next.js?

**Reply by mhsdesign** [reply]
Feel free to look into the examples  https://github.com/remcohaszing/monaco-tailwindcss/tree/main/examples

In the end it’s just like any other js lib ;)
(Of course you should know how to bundle and build stuff)

**Comment by Itskrish01** [comment]
I am not able to find a way to integrate this library with ReactJS or NextJS. Can you guys give me some hint for it? when configuring configureMonacoTailwindcss(monaco, { tailwindConfig });, it shows an error says - "cannot read undefined languages
'"
